File: app/src/main/java/com/k4dima/chat/core/di/ApiModule.java
package com.k4dima.chat.core.di;

import android.support.annotation.NonNull;

import com.k4dima.chat.core.api.LemonadeService;

import org.jetbrains.annotations.NotNull;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

@Module
final class ApiModule {
    @NonNull
    @Singleton
    @Provides
    static LemonadeService lemonadeService(@NotNull Retrofit retrofit) {
        return retrofit.create(LemonadeService.class);
    }

    @NonNull
    @Singleton
    @Provides
    static Retrofit retrofit(OkHttpClient okHttpClient) {
        return new Retrofit.Builder()
                .baseUrl("https://api.lemonade.com/")
                .client(okHttpClient)
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .build();
    }
}

File: app/src/main/java/com/k4dima/chat/core/di/ApiModule.kt
package com.k4dima.chat.core.di

import com.k4dima.chat.core.api.LemonadeService
import dagger.Module
import dagger.Provides
import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory
import javax.inject.Singleton

@Module
internal object ApiModule {
    @JvmStatic
    @Singleton
    @Provides
    fun lemonadeService(retrofit: Retrofit) =
            retrofit.create(LemonadeService::class.java)!!

    @JvmStatic
    @Singleton
    @Provides
    fun retrofit(okHttpClient: OkHttpClient) = Retrofit.Builder()
            .baseUrl("https://api.lemonade.com/")
            .client(okHttpClient)
            .addConverterFactory(GsonConverterFactory.create())
            .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
            .build()!!
}